fe/fe-core/src/main/java/com/starrocks/sql/common/SqlDigestBuilder.java

```diff
@@ -50,6 +92,26 @@ public String visitInPredicate(InPredicate node, Void context) {
             }
         }
 
+        @Override
+        public String visitCompoundPredicate(CompoundPredicate node, Void context) {
+            Pair<Integer, Integer> pair = countCompound(node);
+            if (pair.first + pair.second >= MASSIVE_COMPOUND_LIMIT) {
+                // Only record de-duplicated slots if there are too many compounds
+                Set<SlotRef> exprs = Sets.newHashSet();
+                traverse(node, x -> {
+                    if (x instanceof SlotRef) {
+                        exprs.add((SlotRef) x);
+                    }
+                });
+                return "$massive_compounds[" + exprs.stream().map(SlotRef::toSqlImpl)
```

Comment:
SlotRefs are query-specific so unstable, suggest to use original SlotRefs(slotref references to table column) and convert original slotrefs into table columns then sort these columns lexicographically.

Reply:
good idea.